content/articles/react-flux-frontend.md

```diff
@@ -0,0 +1,65 @@
+
+You might also see the term isomorphic (confusing) or more recently universal, meaning that it can be rendered on the server as well.
+
+
+### Dev dependencies
+If you look at most React/flux boilerplates, you would notice they might use different tools than you might be used with Angular for example.
+
+
+Firstly, as mentioned in the previous section, Babel is omnipresent along with its linter of choice, [eslint](http://eslint.org/).
+
+
+A good number of projects eschew tasks runners such as [gulp](http://gulpjs.com/) in favour of a tool called [Webpack](http://webpack.github.io/) and npm tasks. Webpack is a module bundler like [Browserify](http://browserify.org/) but while Browserify takes the Unix philosophy of having small modules working together, Webpack comes with a lot more out of the box. Both works great though, so choose the one you like the most.
+
+
+One feature that made Webpack stand out early on with React was hot reloading using [react-hot-loader](https://github.com/gaearon/react-hot-loader) but is now available as [LiveReactload](https://github.com/milankinen/livereactload) for browserify.
+
+Testing wise, I have seen lots of repos using [jsdom](https://github.com/tmpvar/jsdom) which seems backwards to me as we can run tests in actual browsers using [Karma](http://karma-runner.github.io/0.13/index.html) for example.
+
+
+### Dependencies
+In terms of actual dependencies, we obviously need React. Note that since 0.14, you will also need the [react-dom](https://www.npmjs.com/package/react-dom) package to actually render.
+
+In terms of Flux implementation, my favourite is [Redux](http://rackt.github.io/redux/) which actually deviates of the Flux reference, leaning towards a FRP (look [here](https://gist.github.com/staltz/868e7e9bc2a7b8c1f754) for a nice introduction on reactive programming) approach like [Elm](http://elm-lang.org/) and having pure functions.
+
+The documentation of Redux being first-class, I will not write about it and invite you to read it, even if you aren't planning to use it.
+Bindings for React can be found under the [react-redux](https://www.npmjs.com/package/react-redux).
+
+A very nice thing about having pure reducers in Redux (the equivalent of stores in Flux) is that you can easily build time-travelling tools and replay actions. Look at [redux-devtools](https://github.com/gaearon/redux-devtools) for the official one.
+
+Lastly there is another optional dependency I like: [ImmutableJS](http://facebook.github.io/immutable-js/). Reducers in Redux are pure but using ImmutableJS gives us a faster `shouldComponentUpdate` (more [info](https://facebook.github.io/react/docs/advanced-performance.html#immutable-js-to-the-rescue)) as we can compare our props by reference rather than inspecting their values deeply.
+
+
+## A boilerplate
+Looking at that list may appear daunting at first but plenty of templates exist but they didn't fit my own needs as I was first writing it in Typescript.
```

Comment:
I'd either use the word "template" or the word "boilerplate" unless there is a distinction I'm not aware of.

Given that your boilerplate is now using Babel, what's its value compared to the other boilerplates?

In any case this sentence is not ideal with the double "but".